src/main/java/org/springframework/data/redis/aot/RedisRuntimeHints.java

```diff
@@ -47,62 +48,90 @@
+						TypeReference.of(ReactiveClusterZSetCommands.class), TypeReference.of(ReactiveClusterHashCommands.class),
+						TypeReference.of(ReactiveClusterServerCommands.class),
+						TypeReference.of(ReactiveClusterStreamCommands.class),
+						TypeReference.of(ReactiveClusterScriptingCommands.class),
+						TypeReference.of(ReactiveClusterGeoCommands.class),
+						TypeReference.of(ReactiveClusterHyperLogLogCommands.class), TypeReference.of(ReactiveRedisOperations.class),
+						TypeReference.of(ReactiveRedisTemplate.class), TypeReference.of(RedisOperations.class),
+						TypeReference.of(RedisTemplate.class), TypeReference.of(StringRedisTemplate.class),
+						TypeReference.of(KeyspaceConfiguration.class), TypeReference.of(MappingConfiguration.class),
+						TypeReference.of(MappingRedisConverter.class), TypeReference.of(RedisConverter.class),
+						TypeReference.of(RedisCustomConversions.class), TypeReference.of(ReferenceResolver.class),
+						TypeReference.of(ReferenceResolverImpl.class), TypeReference.of(IndexConfiguration.class),
+						TypeReference.of(ConfigurableIndexDefinitionProvider.class), TypeReference.of(RedisMappingContext.class),
+						TypeReference.of(RedisRepositoryFactoryBean.class), TypeReference.of(RedisQueryCreator.class),
+						TypeReference.of(MessageListener.class), TypeReference.of(RedisMessageListenerContainer.class),
+
+						TypeReference
+								.of("org.springframework.data.redis.core.BoundOperationsProxyFactory$DefaultBoundKeyOperations"),
+						TypeReference.of("org.springframework.data.redis.core.DefaultGeoOperations"),
+						TypeReference.of("org.springframework.data.redis.core.DefaultHashOperations"),
+						TypeReference.of("org.springframework.data.redis.core.DefaultKeyOperations"),
+						TypeReference.of("org.springframework.data.redis.core.DefaultListOperations"),
+						TypeReference.of("org.springframework.data.redis.core.DefaultSetOperations"),
+						TypeReference.of("org.springframework.data.redis.core.DefaultStreamOperations"),
+						TypeReference.of("org.springframework.data.redis.core.DefaultValueOperations"),
+						TypeReference.of("org.springframework.data.redis.core.DefaultZSetOperations"),
+
+						TypeReference.of(RedisKeyValueAdapter.class), TypeReference.of(RedisKeyValueTemplate.class),
+
+						// Key-Value
+						TypeReference.of(KeySpace.class), TypeReference.of(AbstractKeyValueAdapter.class),
+						TypeReference.of(KeyValueAdapter.class), TypeReference.of(KeyValueOperations.class),
+						TypeReference.of(KeyValueTemplate.class), TypeReference.of(KeyValueMappingContext.class),
+						TypeReference.of(KeyValueRepository.class), TypeReference.of(KeyValueRepositoryFactoryBean.class),
+						TypeReference.of(QueryCreatorType.class), TypeReference.of(KeyValuePartTreeQuery.class)),
+
+				hint -> hint.withMembers(MemberCategory.INVOKE_DECLARED_CONSTRUCTORS, MemberCategory.INVOKE_DECLARED_METHODS));
```

Comment:
Why have you changed from the `INVOKE_PUBLIC_METHODS` to this `INVOKE_DECLARED_METHODS`, please?

Reply:
capturing invocations with the `RuntimeHintsRecorder` indicated we need to have a broader reflection scope.